src/test_driver/linux-cirque/SubscriptionResumptionCapacityTest.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+"""
+Copyright (c) 2024 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import os
+import sys
+
+from helper.CHIPTestBase import CHIPVirtualHome
+
+"""
+Test to verify that the device can still handle new subscription requests when resuming the maximum subscriptions.
```

Comment:
Why does this need to be a cirque test?  It seems this could just as easily be a YAML test or normal python test without involving cirque..

Reply:
Agree, maybe normal python test could achieve this including the device reboot scenario as well, but during testing, we find If using the two controller (devCtrl and devCtrl2) in one same container/host to execute this test, the accessory can still resolve the address for first controller even if the first controller is shutdown by 'self.devCtrl.Shutdown()'. And the server will fail to establish the subscriptions immediately, which makes it hard to send the new subscription request from the second controller, which may be the bug. 

We have added the multiple mobile container to work around it. We would like to gate the persistent subscription quality issue fixed in https://github.com/project-chip/connectedhomeip/pull/30491 asap for future glitch.

Reply:
would follow-up it in https://github.com/project-chip/connectedhomeip/issues/31356, thanks

Reply:
I am not familiar with the YAML test. Is there some way to trigger the server app rebooting and make the server app keep the state of establishing the sessions for a while? We need to send a new subscription during that time.

Reply:
> Is there some way to trigger the server app rebooting

Yes.

> make the server app keep the state of establishing the sessions for a while

I'm not sure what you are asking here @wqx6 .

Reply:
The crash fixed in #30491 will be triggered when the device receives a new subscription during it resuming maximum subscriptions. Normally the time of the device resuming subscriptions is really short. And it is hard to send the new subscription request in the test, I use two controllers for this test and shutdown the first controller after it establish maximum subscription. So when the device reboots, it will keep resolving the address of the first controller for a longer time which is enough for the second controller to send the subscription request.